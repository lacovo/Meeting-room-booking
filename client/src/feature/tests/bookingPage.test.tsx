import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import BookingPage from '../bookingPage';
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

jest.mock('react-redux', () => ({
  useDispatch: () => {},
  useSelector: () => {
    return [{   status: 0,
      match: {roomId: "", roomName: "", date:"", host: "", guest: "", orderBy: "default", is_all: false},
      bookings: [],
      rooms: [], }];
  },
}));

describe('<BookingPage />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = shallow(<BookingPage />);
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });
  it('Should render and match the snapshot', () => {
    const wrapper = shallow(<BookingPage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
