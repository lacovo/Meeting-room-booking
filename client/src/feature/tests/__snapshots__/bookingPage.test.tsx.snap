// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookingPage /> Should render and match the snapshot 1`] = `
<Fragment>
  <Navbar
    title="Booking Table"
  />
  <div>
    <EnhancedTable />
  </div>
</Fragment>
`;
